### ISSUE TITLE
QA/QC for denoised data and connectivity matrices

### ISSUE THREAD

**Issue by celprov** [TheAxonLab/hcph-sops#384]
I propose implementing another QC checkpoint after denoising the data (i.e after nuisance regression and band pass filtering) and estimating the functional connectivity. That report would contain
- [x] ~~Plot the distribution of effective degrees of freedom (cf [Morfini 2023](https://pubmed.ncbi.nlm.nih.gov/37034165/) for the formula how it is computed)~~
- [x] Plot the FC distributions for each session (cf. fig. 10 of Morfini 2023)
- [x] Plot the QC-FC distribution. QC-FC corresponds to the correlation of FC strength with three IQMs (`fd_mean`, `fd_num` and `fd_perc`) and compute QC-FC% which is the distance between the observed distribution and a null distribution generated by permutation (cf table 1 and fig. 11 of Morfini 2023)
- [x] Plot the QC-FC versus euclidean distance separating nodes (cf [Ciric 2017](https://pubmed.ncbi.nlm.nih.gov/28302591/))

**Comment by acionca** [comment]
This could be implemented as part of the visual reports of the fMRI denoising and functional connectivity (see #204 )

**Comment by celprov** [comment]
Yes absolutely! I forgot you already started to set up a visual report for FC.

**Comment by celprov** [comment]
We decided to not compute the effective degrees of freedom (dof) and report its distribution for our study.

Here is the rationale:
It is not clear how Morfini 2023 includes the information about bandpass filtering in the computation of dof. Ciric 2017 do not consider the loss of dof due to the bandpass filtering with the justification that this filter is uniform across all strategies he compares.  As such in Ciric 2017, dof is given by #scans - #regressors - #volumes censored.
As in our study, the #scans and the #regressors stay constant between all sessions, this would leave the dof to be a proxy for #volume censored, so this will duplicate an information that is already computed. Additionally, we do not expect a lot of censored volume as we observed that the FD rarely exceeds 0.2mm. In conclusion, the degrees of freedom is not worth implementing given the specificities of our study.
